src/docker/local.rs

```diff
@@ -121,9 +123,16 @@ pub(crate) fn run(
             .wrap_err("when building custom image")?;
     }
 
-    docker
+    Container::create(engine.clone(), container)?;
+    let status = docker
         .arg(&image_name)
         .args(["sh", "-c", &build_command(toolchain_dirs, &cmd)])
         .run_and_get_status(msg_info, false)
-        .map_err(Into::into)
+        .map_err(Into::into);
+
+    if status.as_ref().map(|x| x.success()).unwrap_or_default() {
```

Comment:
This actually should probably only ignore the specific termination signals handled by `ctrlc`, so anything other than those codes is consider "success" (IE, a termination handler wasn't called).